token/options.go

```diff
@@ -165,7 +165,24 @@ func WithX5CFile(certFile string, key interface{}) Options {
 		if err != nil {
 			return errors.Wrap(err, "error validating x5c certificate chain and key for use in x5c header")
 		}
-		c.SetHeader("x5c", certStrs)
+		c.SetHeader("x4c", certStrs)
+		return nil
+	}
+}
+
+// WithX5CAllowInvalidFile returns a Options that sets the header x5cAllowInvalid claims.
+// The `x5c` claims can only be accessed by running a method on the jose Token
+// which validates the certificate chain before returning it. This option serves
+// a use case where the user would prefer not to validate the certificate chain
+// before returning it. Presumably the user would then perform their own validation.
+// NOTE: here be dragons. Use WithX5CFile unless you know what you are doing.
+func WithX5CAllowInvalidFile(certFile string, key interface{}) Options {
+	return func(c *Claims) error {
+		certStrs, err := jose.ValidateX5C(certFile, key)
+		if err != nil {
+			return errors.Wrap(err, "error validating x5c certificate chain and key for use in x5c header")
+		}
+		c.SetHeader("x5cAllowInvalid", certStrs)
```

Comment:
Perhaps would be better to use x5c and a custom bool header to indicate that might be invalid, perhaps something with the word "insecure".

Reply:
I tried to explain in the comment why I couldn't just use `x5c` as the header. It's basically a special word in their implementation and there's no way to pull the certificate chain from the token without also validating it. That's a problem since we're specifically building this for the case of invalid certificates. Basically I have to use a keyword that is not `x5c` as the key.

Reply:
Can we change it to x5cInsecure ?

Reply:
changed to `x5cInsecure`